src/evict/evict_page.c

```diff
@@ -143,9 +144,22 @@ done:	if (((inmem_split && ret == 0) || (forced_eviction && ret == EBUSY)) &&
  * __wt_evict_page_clean_update --
  *	Update a clean page's reference on eviction.
  */
-void
-__wt_evict_page_clean_update(WT_SESSION_IMPL *session, WT_REF *ref)
+int
+__wt_evict_page_clean_update(
+    WT_SESSION_IMPL *session, WT_REF *ref, int exclusive)
 {
+	int evict;
+
+	/*
+	 * If doing normal system eviction, but only in the service of reducing
+	 * the number of dirty pages, leave the clean page in cache.
+	 */
+	if (!exclusive) {
```

Comment:
This value means two things: first, we have exclusive access to the entire file (we don't need to lock down the page), and second, we want to write, evict and discard the page, we're not interested in in-memory splits or retaining clean pages, all the transactions need to be visible, and so on.

In other words, it's "normal system eviction processing" and everything else (currently salvage and close).

How about renaming "exclusive" to "discard", or "closing"?